### PR TITLE
adds logger in utils

### PR DESCRIPTION
**Pull request by rashidsp**

### CODE REVIEW

File: lib/optimizely/event_builder.rb [hunk 1]
```diff
@@ -50,8 +50,9 @@ class BaseEventBuilder
 
     attr_reader :config
 
-    def initialize(config)
+    def initialize(config, logger)
       @config = config
+      @logger = logger
```
Comment (oakbani) [inline lib/optimizely/event_builder.rb]: Please add logger on the top like config: attr_reader :config

File: lib/optimizely/helpers/event_tag_utils.rb [hunk 2]
```diff
@@ -32,7 +31,7 @@ def string_numeric?(str)
         false
       end
 
-      def get_revenue_value(event_tags)
+      def get_revenue_value(event_tags, logger)
```
Comment (oakbani) [inline lib/optimizely/helpers/event_tag_utils.rb]: Add logger in method comments

File: lib/optimizely/helpers/event_tag_utils.rb [hunk 3]
```diff
@@ -32,7 +31,7 @@ def string_numeric?(str)
...
         # Grab the revenue value from the event tags. "revenue" is a reserved keyword.
         # The revenue value must be an integer.
```
Comment (oakbani) [inline lib/optimizely/helpers/event_tag_utils.rb]: This is not true, now we also accept string/float if they can be parsed to an int without change in value

File: spec/event_tag_utils_spec.rb [hunk 4]
```diff
@@ -21,72 +21,88 @@
...
+      expect(Optimizely::Helpers::EventTagUtils.get_revenue_value(event_tags, logger)).to be_nil
     end
     it 'should return nil if event tags contains the revenue with a string value' do
       event_tags = {
         'revenue' => '65536'
       }
-      expect(Optimizely::Helpers::EventTagUtils.get_revenue_value(event_tags)).to be_nil
+      expect(logger).to receive(:log)
+        .with(Logger::WARN, 'Failed to parse revenue value 65536 from event tags.')
+      expect(Optimizely::Helpers::EventTagUtils.get_revenue_value(event_tags, logger)).to be_nil
```
Comment (oakbani) [inline spec/event_tag_utils_spec.rb]: No, no no. You failed to understand the main purpose of parsing revenue value. Now we do accept strings and float. Please take a look at PHP PR again. See all the changes carefully.

File: lib/optimizely/event_builder.rb (hunk 1 above)
Comment (rashidsp) [reply lib/optimizely/event_builder.rb]: It's not necessary.

**Comment by rashidsp** [comment]
@oakbani implemented revenue parsing!
Please do not merge! 
Unit tests of new method are yet to write.

File: spec/event_tag_utils_spec.rb [hunk 5]
```diff
@@ -21,72 +21,176 @@
 require 'optimizely/helpers/event_tag_utils'
 
 describe 'EventTagUtils' do
-  let(:logger) { Optimizely::NoOpLogger.new }
+  let(:logger) { Optimizely::SimpleLogger.new }
```
Comment (oakbani) [inline spec/event_tag_utils_spec.rb]: We use NoOpLogger in testing. SimpleLogger will output to stdout

File: spec/event_tag_utils_spec.rb [hunk 6]
```diff
@@ -21,72 +21,176 @@
...
         'non-revenue' => 5432
       }
-      expect(Optimizely::Helpers::EventTagUtils.get_revenue_value(event_tags)).to be_nil
+      expect(logger).to receive(:log)
+        .with(Logger::DEBUG, 'The revenue key is not defined in the event tags.')
+      expect(Optimizely::Helpers::EventTagUtils.get_revenue_value(event_tags, logger)).to be_nil
     end
-    it 'should return nil if event tags contains the revenue with a string value' do
+
+    it 'should return nil if event tags contains the revenue key with NULL value' do
```
Comment (oakbani) [inline spec/event_tag_utils_spec.rb]: nil

File: spec/event_tag_utils_spec.rb [hunk 7]
```diff
@@ -21,72 +21,176 @@
...
       event_tags = {
-        'revenue' => 'string'
+        'revenue' => nil
       }
-      expect(Optimizely::Helpers::EventTagUtils.get_revenue_value(event_tags)).to be_nil
+      expect(logger).to receive(:log).with(Logger::DEBUG,
+                                           'The revenue key is null.')
```
Comment (oakbani) [inline spec/event_tag_utils_spec.rb]: key is nil.

File: spec/event_tag_utils_spec.rb [hunk 8]
```diff
@@ -21,72 +21,176 @@
...
+    end
+    it 'should return correct value if event tags contains the revenue with a valid float value' do
+      expect(logger).to receive(:log)
+        .with(Logger::INFO, 'Parsed revenue value 65536 from event tags.')
+      event_tags = {
+        'revenue' => 65_536.0
+      }
+      expect(Optimizely::Helpers::EventTagUtils.get_revenue_value(event_tags, logger)).to eq(65_536)
+    end
+    it 'should return correct value if event tags contains the revenue with a valid string numeric' do
```
Comment (oakbani) [inline spec/event_tag_utils_spec.rb]: nit - make it valid float string

File: spec/event_tag_utils_spec.rb [hunk 9]
```diff
@@ -21,72 +21,176 @@
...
+      expect(logger).to receive(:log)
+        .with(Logger::INFO, 'Parsed revenue value 65536 from event tags.')
+      event_tags = {
+        'revenue' => '65536.0'
+      }
+      expect(Optimizely::Helpers::EventTagUtils.get_revenue_value(event_tags, logger)).to eq(65_536)
+    end
+    it 'should return correct value if event tags contains the revenue with a valid string value' do
```
Comment (oakbani) [inline spec/event_tag_utils_spec.rb]: nit - make it valid integer string

File: spec/notification_center_spec.rb [hunk 10]
```diff
@@ -415,7 +415,7 @@ def deliver_three(_args)
       let(:notification_center) { Optimizely::NotificationCenter.new(spy_logger, raise_error_handler) }
       before(:example) do
         config = Optimizely::ProjectConfig.new(config_body_JSON, spy_logger, error_handler)
-        @event_builder = Optimizely::EventBuilder.new(config)
+        @event_builder = Optimizely::EventBuilder.new(config, @logger)
```
Comment (oakbani) [inline spec/notification_center_spec.rb]: Please pass spy_logger when we are using spy_logger everywhere.

File: spec/event_tag_utils_spec.rb [hunk 11]
```diff
@@ -113,8 +231,8 @@
     end
 
     it 'should return nil if event tags does not contain the numeric tag value' do
-      expect(spy_logger).to receive(:log)
-        .with(Logger::DEBUG, 'The numeric metric key is not defined in the event tags.').exactly(1).times
+      expect(spy_logger).to receive(:log).exactly(1).times
+                                         .with(Logger::DEBUG, 'The numeric metric key is not defined in the event tags.')
```
Comment (oakbani) [inline spec/event_tag_utils_spec.rb]: Put this exactly(1).times after 'with' statement as in all others. 
They will object on this change. No need

**Comment by rashidsp** [comment]
Removes extra PRs

**Pull request closed without merging**